compiler/rustc_parse/src/parser/item.rs

```diff
@@ -707,7 +707,7 @@ impl<'a> Parser<'a> {
             }
             match parse_item(self) {
                 Ok(None) => {
-                    let is_unnecessary_semicolon = !items.is_empty()
+                    let mut is_unnecessary_semicolon = !items.is_empty()
                         // When the close delim is `)` in a case like the following, `token.kind` is expected to be `token::CloseDelim(Delimiter::Parenthesis)`,
                         // but the actual `token.kind` is `token::CloseDelim(Delimiter::Bracket)`.
```

Comment:
```suggestion
                        // but the actual `token.kind` is `token::CloseDelim(Delimiter::Brace)`.
```

I mistakenly wrote `Bracket` for `Brace`. Could you fix that instead?

Reply:
Ok，I didn't notice it, I will rollback my changes on your part.

I also think `TokenTreesReader::parse_token_tree` does a very weird thing for this scenario, I had some try to fix it:
https://github.com/rust-lang/rust/pull/104012#issuecomment-1311764832

currently pending on: https://github.com/rust-lang/rust/pull/104012